src/service/core/workflow/helpers.py

```
@@ -310,6 +310,28 @@ def get_pool_resources(pools: List[str] | None = None,
+    file_name: str,
+) -> storage.LinesStream:
+    """Wrap a storage stream to catch errors and yield them as content.
+
+    Returns the stream wrapped in error handling so that storage errors
+    (e.g. invalid credentials returning 403) are yielded as text content
+    instead of raising an exception that kills the HTTP streaming response.
+    """
+    def _inner() -> Generator[str, None, storage.StreamSummary]:
+        try:
+            yield from stream
+        except Exception as error:  # pylint: disable=broad-exception-caught
```

Ruff (0.15.6)

[warning] Do not catch blind exception: `Exception`

(BLE001)